File: android/UClass/app/src/main/java/com/ubase/uclass/network/response/ChatInitData.kt
package com.ubase.uclass.network.response

import com.google.gson.annotations.SerializedName

/**
 * 채팅 초기화 API 응답 데이터 모델
 */
data class ChatInitData(
    @SerializedName("roomId")
    val roomId: String,

    @SerializedName("branchName")
    val branchName: String,

    @SerializedName("hasMore")
    val hasMore: Boolean,

    @SerializedName("page")
    val page: Int,

    @SerializedName("totalPages")
    val totalPages: Int,

    @SerializedName("totalMessageCount")
    val totalMessageCount: Int,

    @SerializedName("messages")
    val messages: List<ChatMessage>
)

File: ios/uclass/uclass/network/response/ChatInitData.swift
import Foundation

// MARK: - 채팅 초기화 API 응답 데이터 모델
struct ChatInitData: Codable {
    let roomId: String
    let branchName: String
    let hasMore : Bool
    let messages: [ChatMessage]
    
    enum CodingKeys: String, CodingKey {
        case roomId
        case branchName
        case messages
        case hasMore
    }
}
